litex/soc/interconnect/csr.py

```diff
@@ -122,6 +122,66 @@ def do_finalize(self, busword):
         raise NotImplementedError
 
 
+class CSRField(Signal):
+    def __init__(self, name, size=1, offset=None, reset=0, description=None, pulse=False, access=None, values=None):
+        assert name == name.lower()
+        assert access in [None, "write-only", "read-only", "read-write"]
```

Comment:
Would it be better to use an enum here, or a text field as here?  I'm unfamiliar with Python Best Practices.

Valid SVD values are (from https://siliconlabs.github.io/Gecko_SDK_Doc/CMSIS/SVD/html/group__svd__xml__fields__gr.html):

```
Describe the manipulation of data written to a field. If not specified, the value written to the field is the value stored in the field. The other options are bitwise operations:

    oneToClear: write data bit of one shall clear (set to zero) the corresponding bit in the field.
    oneToSet: write data bit of one shall set (set to one) the corresponding bit in the field.
    oneToToggle: write data bit of one shall toggle (invert) the corresponding bit in the field.
    zeroToClear: write data bit of zero shall clear (set to zero) the corresponding bit in the field.
    zeroToSet: write data bit of zero shall set (set to one) the corresponding bit in the field.
    zeroToToggle: write data bit of zero shall toggle (invert) the corresponding bit in the field.
    clear: after a write operation all bits in the field are cleared (set to zero).
    set: after a write operation all bits in the field are set (set to one).
    modify: after a write operation all bit in the field may be modified (default).
```

However, this is such a minor thing that I don't think it should hold up adoption of this patchset.

Reply:
In fact i'm not sure to understand what is your intend for values: do you want to store possible values of the field + description as you did in https://github.com/xobs/dreg/blob/master/dcsr.py#L290-L301 or the manipulation of data in the field as seem to describe the modifiedWriteValues SVD values? I think both could be useful, but maybe in a first time we should just support what you intended to do in dcsr?

Reply:
Both.

`values` stores the possible values, plus each description of that value.  Later on we may want to add some validation, ensuring the values are actual valid numbers.  But either way, this describes what the contents of the register can be, and what each value means.

`access` indicates what the value does.  For example, the bits in the `EventManager.pending` are all `oneToClear`, since writing a 1 to that CSR clears the `pending` bit.

The only additional fields we may want to add are `min` and `max`, which would define ranges (such as for clock dividers).

Reply:
Thanks, i agree on the features, maybe i'll first merge the rest of the code and will open another pull-request to discuss this more.